docs/architecture/maui/mvvm.md

```diff
@@ -0,0 +1,364 @@
+
+Model classes are typically used in conjunction with services or repositories that encapsulate data access and caching.
+
+## Connecting view models to views
+
+View models can be connected to views by using the data-binding capabilities of .NET MAUI. There are many approaches that can be used to construct views and view models and associate them at runtime. These approaches fall into two categories, known as view first composition, and view model first composition. Choosing between view first composition and view model first composition is an issue of preference and complexity. However, all approaches share the same aim, which is for the view to have a view model assigned to its BindingContext property.
+
+With view first composition the app is conceptually composed of views that connect to the view models they depend on. The primary benefit of this approach is that it makes it easy to construct loosely coupled, unit testable apps because the view models have no dependence on the views themselves. It's also easy to understand the structure of the app by following its visual structure, rather than having to track code execution to understand how classes are created and associated. In addition, view first construction aligns with the Microsoft Maui's navigation system that's responsible for constructing pages when navigation occurs, which makes a view model first composition complex and misaligned with the platform.
+
+With view model first composition, the app is conceptually composed of view models, with a service responsible for locating the view for a view model. View model first composition feels more natural to some developers, since the view creation can be abstracted away, allowing them to focus on the logical non-UI structure of the app. In addition, it allows view models to be created by other view models. However, this approach is often complex, and it can become difficult to understand how the various parts of the app are created and associated.
+
+> [!TIP]
+> Keep view models and views independent
+
+The binding of views to a property in a data source should be the view's principal dependency on its corresponding view model. Specifically, don't reference view types, such as Button and ListView, from view models. By following the principles outlined here, view models can be tested in isolation, therefore reducing the likelihood of software defects by limiting scope.
+
+The following sections discuss the main approaches to connecting view models to views.
+
+## Creating a view model declaratively
+
+The simplest approach is for the view to declaratively instantiate its corresponding view model in XAML. When the view is constructed, the corresponding view model object will also be constructed. This approach is demonstrated in the following code example:
+
+```xaml
+<ContentPage ... xmlns:local="clr-namespace:eShop">
+    <ContentPage.BindingContext>
+        <local:LoginViewModel />
+    </ContentPage.BindingContext>
+    ...
```

Comment:
More often than not, readers will copy code verbatim from wherever they find it. Please try to avoid putting code in place that will not compile. However, if you do need to have something that indicates code was omitted, use a valid (and language-appropriate) comment. For example:

**C#**

```csharp
public sealed record struct Point(double X, double Y)
{
    // Omitted for brevity...
}
```

**XAML**

```xaml
<ContentPage>
    <!-- Omitted for brevity... -->
</ContentPage>
```

Reply:
That is no problem. I wasn't sure what the standard was there. I will follow this going forward though.